### PR TITLE
Dt49 glossary auto link r1

### PR DESCRIPTION
**Pull request by coreysals**
Issues:
https://alliancels.myjetbrains.com/youtrack/issue/DT-49
https://alliancels.myjetbrains.com/youtrack/issue/TIT-144

Doc tools changes so that glossary terms can be automagically linked. 

Links are inserted in documentation by placing exclamation point in front and behind term (eg if you wanted "Ready" to be linked you'd type "!Ready!".

Currently the auto linking is done as part of the "combineAll" command/task. This can be changed in the future if desired but I figured it would be better to keep it separate as it might add build time to the "aA" command/task. 

I'm looking for any/all constructive criticism so that these changes can be helpful to the group!

Let me know if you have any questions.

Thanks!

In order to test this on the documentation-gradle-plugins example project:
-Checkout this branch
-In cmd.exe go to WorkingDirectory/documentation-gradle-plugins/example
-Run "gradlew --include-build .. aA --rerun-tasks" command
-Run "gradlew --include-build .. combineAll" command
-Check glossary terms and glossary term links throughout example documentation and maybe add some of your own!

In order to test this on Titan documentation:
-Checkout https://github.com/alliancels/Titan/tree/TIT144_AutoLinkGlossaryDocTools
-In cmd.exe go to WorkingDirectory/Titan
-Run "gradlew --refresh-dependencies" command
-Run "gradlew Docs:aA --rerun-tasks" command
-Run "gradlew Docs:combineAll" command
-Check glossary terms and glossary term links in documentation and maybe add some of your own!

### CODE REVIEW

File: src/main/groovy/com/alliancels/documentation/GlossaryAutoLink.groovy [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            println("srcString: " + srcString)
+            println("dstString: " + dstString)
+            
+            String eachHypertextString = hypString - srcString
+            println("Hypertext file postfix: " + eachHypertextString) 
+            
+            eachHypertextString = dstString + "/documentation/All" + eachHypertextString
+            println("Hypertext file: " + eachHypertextString)
+            
+            //eachHypertextString = eachHypertextString + "/section.html"
```
Comment (nicolebartel) [inline src/main/groovy/com/alliancels/documentation/GlossaryAutoLink.groovy]: Can delete these 2 lines.

**Review by nicolebartel — APPROVED** [review]

**Comment by coreysals** [comment]
Things added/changed:
-Update all links in combined docs to jump-to links
-Add auto linking to the glossary itself
-Remove link/location and anchor columns from glossary table
-Smaller bug fixes, cleanup, and comments

**Comment by coreysals** [comment]
The changes have also been brought to the maven repo here:
https://github.com/alliancels/maven-repo/tree/DT49_ExternalTestBranch

So you can use this (the same as before) branch to test changes on Titan:
https://github.com/alliancels/Titan/tree/TIT144_AutoLinkGlossaryDocTools

**Review by jdoutine — APPROVED** [review]
Latest changes look good. Thanks for implementing my feedback!

**Pull request merged**